types.ts

```diff
@@ -413,6 +418,30 @@ export interface WebBasicAuthentication extends BaseAuthentication {
   };
 }
 
+export interface CustomAuthParameter {
+  /**
+   * The name used to refer to the parameter. Should match the string for the key in the params mapping.
+   */
+  name: string;
+
+  /**
+   * A description shown to the user indicating what value they should provide for this parameter.
+   */
+  description: string;
+}
+
+/**
+ * Authenticate using custom templated parameters, which you insert into the request. The user provides all the
+ * parameters you specify, and you have access to them through a templating engine.
+ */
+export interface CustomAuthentication extends BaseAuthentication {
+  type: AuthenticationType.Custom;
+  /**
+   * Configuration for labels to show in the UI when the user sets up a new acount.
```

Comment:
This would be more than just UI labels, it defines what parameters are eligible for use at all.

Reply:
yeah good call this was an old string. updated to be more descriptive